app/src/molecules/DeckThumbnail/index.tsx

```diff
@@ -69,16 +69,18 @@ export function DeckThumbnail(props: DeckThumbnailProps): JSX.Element {
               ? initialLoadedLabwareBySlot[slotId]
               : null
           const labwareInModule =
-            moduleInSlot &&
+            moduleInSlot?.result?.moduleId != null &&
             moduleInSlot.result.moduleId in initialLoadedLabwareByModuleId
-              ? initialLoadedLabwareByModuleId[moduleInSlot.result.moduleId]
+              ? initialLoadedLabwareByModuleId[moduleInSlot.result?.moduleId]
```

Comment:
nit: you already type guarded in the and statement above so i don't think you need the question mark

```suggestion
              ? initialLoadedLabwareByModuleId[moduleInSlot.result.moduleId]
```